documentation/src/main/asciidoc/search-dsl-sort.asciidoc

```diff
@@ -208,3 +193,132 @@ include::{sourcedir}/org/hibernate/search/documentation/search/sort/SortDslIT.ja
+
+Documents that do not have any value for a sort field will be assigned a default value
+depending on the sort and field type (`0`, an empty string, ...).
+
+The behavior for missing values can be controlled explicitly through the `.missing()` option:
+
+* `.missing().first()` puts documents with no value in first position (regardless of the sort order).
+* `.missing().last()` puts documents with no value in last position (regardless of the sort order).
+* `.missing().use(...)` uses the given value as a default for documents with no value.
+
+Below are a few examples with the <<search-dsl-sort-field,field sort>>.
+
+.Sorting by field values, documents with no value in first position
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/sort/SortDslIT.java[tags=missing-first]
+----
+====
+
+.Sorting by field values, documents with no value in last position
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/sort/SortDslIT.java[tags=missing-last]
+----
+====
+
+.Sorting by field values, documents with no value using a given default value
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/sort/SortDslIT.java[tags=missing-use]
+----
+====
+
+[[search-dsl-sort-common-multi-value-mode]]
+=== Sort mode for multi-valued fields
+
+Documents that have multiple values for a sort field can be sorted too.
+A single value is picked for each document in order to compare it with order documents.
+How the value is picked is called the *sort mode*, specified using the `.mode(...)` option.
+The following sort modes are available:
+
+|====
+|Mode|Description|Supported value types|Unsupported value types
+|`SortMode.MIN`
+|Picks the lowest value for field sorts, the lowest distance for distance sorts.
+
+This is default for ascending sorts.
+|All.
+|-
+|`SortMode.MAX`
+|Picks the highest value for field sorts, the highest distance for distance sorts.
+
+This is default for descending sorts.
+|All.
+|-
+|`SortMode.SUM`
+|Computes the sum of all values for each document,
+and picks that sum for comparison with other documents.
+|Numeric fields (`long`, ...).
+|Text and temporal fields (`String`, `LocalDate`, ...), <<search-dsl-sort-distance,distance>>.
+|`SortMode.AVG`
+|Computes the https://en.wikipedia.org/wiki/Arithmetic_mean[arithmetic mean] of all values for each document
+and picks that average for comparison with other documents.
+|Numeric and temporal fields (`long`, `LocalDate`, ...), <<search-dsl-sort-distance,distance>>.
+|Text fields (`String`, ...).
+|`SortMode.MEDIAN`
+|Computes the https://en.wikipedia.org/wiki/Median[median] of all values for each document,
+and picks that median for comparison with other documents.
+|Numeric and temporal fields (`long`, `LocalDate`, ...), <<search-dsl-sort-distance,distance>>.
+|Text fields (`String`, ...).
```

Comment:
Maybe in the future we could support the median of a string keyword field.

Reply:
It's not supported by Elasticsearch. I didn't check, but the reason is probably about performance. But yeah, maybe, in the future, if someone wants it.